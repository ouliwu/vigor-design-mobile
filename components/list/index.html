<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"
    />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, viewport-fit=cover"
    />
    <meta
      name="keywords"
      content="vigor-design-mobile, Vigor Design Mobile , React mobile components"
    />
    <meta
      name="description"
      content="Essential UI blocks for building mobile web apps."
    />
    <meta
      name="google-site-verification"
      content="2saDOufpfJI4y1cqfxvuviYg8bfo4gLmiEtSnt2oh50"
    />
    <link
      rel="shortcut icon"
      type="image/x-icon"
      href="https://user-images.githubusercontent.com/48983788/199143833-07d9c5d9-6ddf-4d54-b6d2-8530fea2dec3.svg"
    />
    <script>
      window.routerBase = "/vigor-design-mobile";
    </script>
    <script>
      //! umi version: 3.5.35
    </script>
    <script>
      !(function () {
        var e =
            navigator.cookieEnabled && void 0 !== window.localStorage
              ? localStorage.getItem("dumi:prefers-color")
              : "auto",
          o = window.matchMedia("(prefers-color-scheme: dark)").matches,
          t = ["light", "dark", "auto"];
        document.documentElement.setAttribute(
          "data-prefers-color",
          e === t[2] ? (o ? t[1] : t[0]) : t.indexOf(e) > -1 ? e : t[0]
        );
      })();
    </script>
    <title>List 列表 - Vigor Design Mobile</title>
  </head>
  <body>
    <div id="root"></div>

    <script>
      if (location.pathname.startsWith("/~demos/")) {
        document.documentElement.setAttribute("data-is-demo", "true");
      }
    </script>
    <script>
      if (!location.port) {
        // Enable Google Analytics
        window.dataLayer = window.dataLayer || [];
        function gtag() {
          dataLayer.push(arguments);
        }
        gtag("js", new Date());
        gtag("config", "UA-72788897-2");
      }
    </script>
    <script
      src="https://www.googletagmanager.com/gtag/js?id=UA-72788897-2"
      async="true"
    ></script>
    <script src="/vigor-design-mobile/umi.dfd80f39.js"></script>
  </body>
</html>
