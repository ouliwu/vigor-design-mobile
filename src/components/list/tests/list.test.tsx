import React, { createRef } from 'react'
import { render, testA11y } from 'testing'
import List from '..'
import { ListRef } from '../list'

describe('list', () => {
  test('a11y', async () => {
    await testA11y(
      <List>
        <List.Item>1</List.Item>
      </List>
    )
  })

  test('should works given ref', async () => {
    const ref = createRef<ListRef>()
    const { baseElement } = render(
      <List ref={ref}>
        <List.Item>1</List.Item>
      </List>
    )
    expect(ref.current).toBeDefined()
    expect(ref.current?.nativeElement).toBeDefined()
    expect(baseElement).toMatchSnapshot()
  })
})
