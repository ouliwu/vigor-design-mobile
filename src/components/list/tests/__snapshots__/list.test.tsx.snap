// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should works given ref 1`] = `
<body>
  <div>
    <div
      class="adm-list adm-list-default"
    >
      <div
        class="adm-list-body"
      >
        <div
          class="adm-list-body-inner"
        >
          <div
            class="adm-list-item"
          >
            <div
              class="adm-list-item-content"
            >
              <div
                class="adm-list-item-content-main"
              >
                1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
